src/Microsoft.DotNet.Build.Tasks/GetNetCoreAppVersionsFromFile.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GetNetCoreAppVersionsFromFile : Task
+    {
+        [Required]
+        public string PathToVersionsFile { get; set; }
+
+        [Output]
+        public string CorefxHash { get; set; }
```

Comment:
I think I would make this a little more generic so that we can easily extend it to things like standard version or others as well. Instead of returning hard-coded properties, return an ITaskItem[] and add set of items which have identity of the key and metadata of the sha.